src/providers.dev.net.spec.ts

```diff
@@ -46,59 +54,60 @@ describe("test network providers on devnet: Proxy and API", function () {
-                let apiResponse = await apiProvider.getFungibleTokenOfAccount(user, item.identifier);
-                let proxyResponse = await proxyProvider.getFungibleTokenOfAccount(user, item.identifier);
-
-                assert.deepEqual(apiResponse, proxyResponse, `user: ${user.bech32()}, token: ${item.identifier}`);
+            for (let i = 0; i < apiResponse.length; i++) {
+                assert.equal(apiResponse[i].identifier, proxyResponse[i].identifier);
+                assert.equal(apiResponse[i].balance.valueOf, proxyResponse[i].balance.valueOf);
             }
         }
     });
 
     it("should have same response for getNonFungibleTokensOfAccount(), getNonFungibleTokenOfAccount", async function () {
         this.timeout(30000);
 
-        for (const user of [alice, bob, carol, dan]) {
-            let apiResponse = await apiProvider.getNonFungibleTokensOfAccount(user);
-            let proxyResponse = await proxyProvider.getNonFungibleTokensOfAccount(user);
+        let apiResponse = (await apiProvider.getNonFungibleTokensOfAccount(dan)).slice(0, 20);
+        let proxyResponse = (await proxyProvider.getNonFungibleTokensOfAccount(dan)).slice(0, 20);
 
-            for (const item of apiResponse) {
-                // Gateway does not provide "type".
-                item.type = "";
-            }
+        assert.equal(apiResponse.length, proxyResponse.length);
 
-            assert.deepEqual(apiResponse.slice(0, 100), proxyResponse.slice(0, 100));
+        for (let i = 0; i < apiResponse.length; i++) {
+            removeInconsistencyForNonFungibleTokenOfAccount(apiResponse[i], proxyResponse[i]);
+        }
 
-            for (const item of apiResponse.slice(0, 5)) {
-                let apiResponse = await apiProvider.getNonFungibleTokenOfAccount(user, item.collection, item.nonce);
-                let proxyResponse = await proxyProvider.getNonFungibleTokenOfAccount(user, item.collection, item.nonce);
+        assert.deepEqual(apiResponse, proxyResponse);
 
-                // Gateway does not provide "type".
-                apiResponse.type = "";
+        const item = apiResponse[0];
+        let apiItemResponse = await apiProvider.getNonFungibleTokenOfAccount(dan, item.collection, item.nonce);
+        let proxyItemResponse = await proxyProvider.getNonFungibleTokenOfAccount(dan, item.collection, item.nonce);
 
-                assert.deepEqual(apiResponse, proxyResponse, `user: ${user.bech32()}, token: ${item.identifier}`);
-            }
-        }
+        removeInconsistencyForNonFungibleTokenOfAccount(apiItemResponse, proxyItemResponse);
+        assert.deepEqual(apiResponse, proxyResponse, `user: ${dan.bech32()}, token: ${item.identifier}`);
     });
 
+    // TODO: Strive to have as little differences as possible between Proxy and API.
+    function removeInconsistencyForNonFungibleTokenOfAccount(apiResponse: NonFungibleTokenOfAccountOnNetwork, proxyResponse: NonFungibleTokenOfAccountOnNetwork) {
```

Comment:
Usually when you check the equality between 2 objects that have almost the same type but with some differences, is better to assert only the targeted fields. 

Also, if you want to go with your approach, maybe is better to delete de unnecessary fields from the objects instead of resetting with a default.

eg.

```
delete apiResponse.type;
delete proxyResponse.type;
delete apiResponse.name;
...
```

Reply:
I agree that it's better to assert only for the targeted fields, but the object has many fields so it was easier to just reset the inconsistent fields and then use `assert.deepequal()` than to compare each field. 

I've set the fields to their default value instead of deleting them to be consistent with the function `ignoreKnownTransactionDifferencesBetweenProviders()`